x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchResponse.java

```diff
@@ -512,7 +409,6 @@ public Hits(StreamInput in) throws IOException {
             }
             events = in.readBoolean() ? in.readList(Event::new) : null;
             sequences = in.readBoolean() ? in.readList(Sequence::new) : null;
-            counts = in.readBoolean() ? in.readList(Count::new) : null;
```

Comment:
This will break requests in mixed clusters, are we ok with that?

Reply:
Thanks for flagging this.
EQL is currently experimental so the BWC compatibility is loose however I'd like where possible, to avoid this situation. I've added this topic to the agenda since I can see other areas where this *might* affect us (like using fields API) and we could use more testing.

Reply:
Yeah, I am not sure that experimental necessarily means we don't support rolling restarts. My interpretation was that we can break it if there is a compelling reason for it. So, I would lean towards supporting it whenever possible. Two things we need to keep in mind: 1) if we will ever promote it to non-experimental in some 7.x release this might break rolling upgrade in 7.x. and 2) if we will promote it sometime in 8.x, we should be careful about breaking BWC with 7.last.

Reply:
I've added a compatibility branch just to be safe. We never created `Counter` there's no actual instance to be read, just the bool which should be `false` always.
Nevertheless I've added the previous code since it was small.